public/js/components/Frames.js

```diff
@@ -60,40 +46,66 @@ const Frames = React.createClass({
     });
   },
 
+  renderFrame(frame) {
+    const { selectedFrame, selectFrame } = this.props;
+
+    const selectedClass = (
+      selectedFrame && (selectedFrame.id === frame.id ? "selected" : "")
+    );
+
+    return dom.li(
+      { key: frame.id,
+        className: `frame ${selectedClass}`,
+        onMouseDown: () => selectFrame(frame),
+        tabIndex: 0
+      },
+      renderFrameTitle(frame),
+      renderFrameLocation(frame)
+    );
+  },
+
+  renderFrames() {
+    let { frames } = this.props;
+
+    const numFramesToShow = this.state.showAllFrames
+      ? frames.size : NUM_FRAMES_SHOWN;
+    frames = frames.slice(0, numFramesToShow);
+
+    return dom.ul({}, frames.map(frame => this.renderFrame(frame)));
+  },
+
+  renderToggleButton() {
+    const { frames } = this.props;
+    let buttonMessage = this.state.showAllFrames
+      ? L10N.getStr("callStack.collapse") : L10N.getStr("callStack.expand");
+
+    if (frames.size < NUM_FRAMES_SHOWN) {
+      return null;
+    }
+
+    return dom.div(
+      { className: "show-more", onClick: this.toggleFramesDisplay },
+      buttonMessage
+    );
+  },
+
   render() {
-    const { frames, selectedFrame, selectFrame } = this.props;
-    const numFramesToShow = this.state.showAllFrames ? frames.length : 7;
-    let framesDisplay;
+    const { frames } = this.props;
 
     if (frames.length === 0) {
```

Comment:
Here it is: https://github.com/devtools-html/debugger.html/blob/master/public/js/reducers/pause.js#L140

```javascript
function getFrames(state) {
  return state.pause.get("frames") || [];
}
```

Should be
```javascript
const { Seq } = require("immutable");
// ...
function getFrames(state) {
  return state.pause.get("frames") || Seq();
}
```